torch/_inductor/scheduler.py

```diff
@@ -4049,6 +4039,100 @@ def compute_graph_partition_maps(
                 )
             )
 
+    def get_graph_partition_symbol_inputs(
+        self,
+        partition: PartitionType,
+        input_nodes: dict[str, Union[ir.IRNode, ir.TorchBindObject, sympy.Expr]],
+    ) -> OrderedSet[sympy.Symbol]:
+        """
+        Returns all symbol inputs which are required to be in scope to successfully
+        perform codegen for this graph partition, including:
+        - free symbols used in partition nodes
+        - free symbols in partition input/node shapes, strides, and offsets. This is needed
+          for recording cudagraphs for tensors with dynamic shapes.
+        """
+
+        def get_layout_symints(node: ir.IRNode) -> OrderedSet[sympy.Symbol]:
+            free_symbol_uses: OrderedSet[sympy.Symbol] = OrderedSet()
+            layout = node.maybe_get_layout()
+            if isinstance(layout, ir.Layout):
+                free_symbol_uses.update(
+                    free_symbols(layout.size)
+                    | free_symbols(layout.stride)
+                    | free_symbols(layout.offset)
+                )
+            if isinstance(layout, ir.MutationLayoutSHOULDREMOVE):
+                # symint may be used as index in layout.target
+                free_symbol_uses.update(get_layout_symints(layout.target))
+            return free_symbol_uses
+
+        def get_scheduler_node_symbol_uses(
+            node: BaseSchedulerNode,
+        ) -> OrderedSet[sympy.Symbol]:
+            """
+            Gets symbols used in node.
+            """
+            if isinstance(node, FusedSchedulerNode):
+                return OrderedSet().union(
+                    *(get_scheduler_node_symbol_uses(snode) for snode in node.snodes)
+                )
+            assert node.node is not None
+            free_symbol_uses = node.node.get_free_symbol_uses()
+            free_symbol_uses.update(
+                *(get_layout_symints(ir_node) for ir_node in node.node.get_outputs())
+            )
+            return free_symbol_uses
+
+        def get_input_node_symbols(
+            node: Union[ir.IRNode, sympy.Expr, ir.TorchBindObject],
+        ) -> OrderedSet[sympy.Symbol]:
+            """
+            Gets symbols used in input node shapes, strides, and offsets.
+            """
+            if isinstance(node, ir.TorchBindObject):
+                return OrderedSet()
+            elif isinstance(node, ir.IRNode):
+                return get_layout_symints(node)
+            elif isinstance(node, sympy.Expr):
+                return OrderedSet(node)
```

Comment:
is this being tested? OrderedSet(sympy.Expr) should throw. it takes in an iterable:

```
>>> import sympy
>>> from torch.utils._ordered_set import OrderedSet
>>> OrderedSet(sympy.Symbol("5"))
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "/data/users/eellison/pytorch/torch/utils/_ordered_set.py", line 27, in __init__
    self._dict = dict.fromkeys(iterable, None) if iterable is not None else {}
TypeError: 'Symbol' object is not iterable
```